runtime/common/src/oracle.rs

```diff
@@ -142,3 +148,90 @@ pub mod benchmarks_util {
 		}
 	}
 }
+
+#[derive(Clone, RuntimeDebugNoBound, TypeInfo, Encode, Decode, MaxEncodedLen)]
+#[scale_info(skip_type_params(O))]
+pub struct Feeder<O: OriginTrait>(O::PalletsOrigin);
+
+impl<O: OriginTrait> PartialEq for Feeder<O> {
+	fn eq(&self, other: &Self) -> bool {
+		self.0.eq(&other.0)
+	}
+}
+
+impl<O: OriginTrait> Eq for Feeder<O> {}
+
+#[cfg(feature = "runtime-benchmarks")]
+impl<O: OriginTrait<AccountId = AccountId>> From<u32> for Feeder<O> {
+	fn from(value: u32) -> Self {
+		Self(O::signed(frame_benchmarking::account("feeder", value, 0)).into_caller())
+	}
+}
+
+/// Get the decimals for the pool currency
+pub fn decimals_for_pool<Pools, AssetRegistry>(pool_id: PoolId) -> Result<u32, DispatchError>
+where
+	Pools: PoolInspect<AccountId, CurrencyId, PoolId = PoolId>,
+	AssetRegistry: asset_registry::Inspect<AssetId = CurrencyId, CustomMetadata = CustomMetadata>,
+{
+	let currency = Pools::currency_for(pool_id).ok_or(DispatchError::Other(
+		"OracleConverterBridge: No currency for pool",
+	))?;
+
+	let metadata = AssetRegistry::metadata(&currency).ok_or(DispatchError::Other(
+		"OracleConverterBridge: No metadata for currency",
+	))?;
+
+	Ok(metadata.decimals)
+}
+
+/// A provider bridge that transform generic quantity representation of a price
+/// into a balance denominated in a pool currency.
+pub struct OracleConverterBridge<Origin, Provider, Pools, AssetRegistry>(
+	PhantomData<(Origin, Provider, Pools, AssetRegistry)>,
+);
```

Comment:
Don't we also need a converter for the `CachedCollection`, or is it implicitly using this one?

Reply:
yes it is implicitly using this one